### PR TITLE
Add sent mail listing

### LINKED ISSUE #469: Add a Sent / outbound listing (search and box cannot list mail you sent)
## Context

There is no first-class way to list **outbound** mail from the CLI.

- `hey box view` documents names like `imbox` / `feedbox`. `paper_trail` / `papertrail` are not accepted as box names (or the trail box lists as empty).
- `hey search --in papertrail` does not surface a usable sent archive for scripting.
- `hey search --from` with the address you actually used on `hey compose --from` often returns nothing (see follow-up issue on `--from` semantics).

The workable workaround is `hey search --to <one recipient>` and repeating that for every address. That does not scale, and it is not how “show me what I sent this week” should work.

No account-specific details below.

## Ask

Expose outbound mail as a listable surface, any of:

1. `hey sent` / `hey box view sent` (or document and implement the real Paper Trail box name + contents that include **your** sends).
2. `hey search --in sent` (or `--outbound`) with the same `--date`, `--subject`, `--to`, `--from`, `--all`, `--json` flags search already has.

JSON rows should include thread id, subject, recipients, sent time, and app url so a script can diff a local send log against HEY.

## Why it matters

After a batch of `hey compose` calls, the only way to reconcile “did these actually leave?” is N recipient searches. A sent list is the missing half of compose.

### PR DESCRIPTION
**Pull request by robzolkos**
## Summary

- add first-class `hey sent` with one latest outbound message per thread, newest first
- match HEY's compact human recipient summary while returning exact `to`, `cc`, and `bcc` arrays in JSON
- include thread ID, subject, summary, delivery time, and app URL for reconciliation scripts
- support `--limit`, `--all`, `--page`, `--ids-only`, `--count`, Markdown, and the standard JSON formats
- document the command and cover its output, pagination, limits, help surface, and smoke contract

Closes #469.

## Dependencies

- [haystack#8874](https://github.com/basecamp/haystack/pull/8874) adds recipients, delivery time, and Sent pagination headers
- [hey-sdk#222](https://github.com/basecamp/hey-sdk/pull/222) models those fields and exposes paginated Sent reads

This branch pins the SDK PR commit with a Go pseudo-version so CI and review are reproducible. It must be changed to the published SDK release before this PR merges.

## Tests

- `make test`
- `make lint`
- `make build`
- `go test -c` for `tests/smoke`
- built `./bin/hey` against a real HEY account through a redacting validator: three rows checked; thread ID, subject, sent time, app URL, and array shapes valid; no message content or personal data printed or retained

Production does not expose recipient data until the Haystack dependency deploys, so that real-account check correctly found no populated recipient arrays. Unit and smoke contracts cover the new response shape.

---
## Summary by cubic
Adds `hey sent`, a first-class command that lists the latest outbound message in each thread, newest first. It provides thread and delivery metadata for scripts while matching HEY's compact human-readable recipient summary. Closes #469.

**New Features**
- JSON includes exact `to`, `cc`, and `bcc` arrays, thread ID, subject, summary, delivery time, and app URL; older servers report `sent_at` as `null`.
- Supports `--limit` (reading pages until the requested count is reached), `--all`, `--page`, `--ids-only`, `--count`, Markdown, and the standard JSON formats.
- `--all` follows pagination for up to 100 pages; notices suggest `--all` when more pages remain and a sanitized `--page` cursor when the run was truncated.
- Documentation, API coverage, help output, unit tests, and smoke coverage include the new command and its edge cases.

**Dependencies**
- Pins `github.com/basecamp/hey-sdk/go` to a commit pseudo-version; replace it with the published SDK release before merging.
- Recipient arrays remain empty until HEY's backend exposes Sent recipient data.

<sup>Written for commit 75fe8f3076d2051eafa8b1c852a87db10fd03695. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/basecamp/hey-cli/pull/473?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: internal/cmd/sent.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  hey sent --limit 10 --json
+  hey sent --all --json
+  hey sent --ids-only`,
+		RunE: sentCommand.run,
+		Args: cobra.NoArgs,
+	}
+
+	sentCommand.cmd.Flags().IntVar(&sentCommand.limit, "limit", 0, "Maximum number of sent messages to show")
+	sentCommand.cmd.Flags().BoolVar(&sentCommand.all, "all", false, "Fetch up to 100 results pages (override --limit)")
+	sentCommand.cmd.Flags().IntVar(&sentCommand.page, "page", 1, "Results page")
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/cmd/sent.go]: Fixed - --page now preserves opaque string cursors and coverage includes a nonnumeric cursor.

File: go.mod [hunk 2]
```diff
@@ -8,7 +8,7 @@ require (
 	charm.land/glamour/v2 v2.0.1
 	charm.land/lipgloss/v2 v2.0.6
 	github.com/basecamp/actioncable-go v1.1.0
-	github.com/basecamp/hey-sdk/go v0.31.1
+	github.com/basecamp/hey-sdk/go v0.31.2-0.20260920180340-2d40721c770f
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply go.mod:11]: Flagged for human review - this dependency pin must remain until SDK PR #222 is merged and released. The PR description records the required release replacement and dependency order.

File: internal/cmd/sent.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+}
+
+func makeSentMessages(topics []generated.Topic) []sentMessage {
+	messages := make([]sentMessage, 0, len(topics))
+	for _, topic := range topics {
+		entry := topic.LatestEntry
+		sentAt := entry.ActiveAt
+		if sentAt.IsZero() {
+			sentAt = entry.CreatedAt
+		}
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/cmd/sent.go]: Fixed - missing active_at now produces an explicit null sent_at and Unavailable in human output, without using creation time.

File: internal/cmd/sent.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+	if len(recipients) == 1 {
+		return "Me → " + name
+	}
+	return fmt.Sprintf("Me → %s + %d", name, len(recipients)-1)
+}
+
+func sentListingNotice(shown, pages int, nextPage string, truncated bool) string {
+	switch {
+	case truncated:
+		return fmt.Sprintf("Sent listing stopped after %d pages. Continue with --page %s.", pages, nextPage)
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/cmd/sent.go]: Fixed - continuation notices sanitize the cursor while raw cursor metadata and requests stay unchanged.

File: internal/cmd/sent.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+	format := writer.EffectiveFormat()
+	if stderrNotice := paginationNoticeForStderr(format, notice); stderrNotice != "" {
+		fmt.Fprintln(cmd.ErrOrStderr(), stderrNotice)
+	}
+
+	switch format {
+	case output.FormatStyled:
+		return writeSentStyled(cmd, messages, notice)
+	case output.FormatMarkdown:
+		return writeOK(makeSentTableRows(messages))
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/cmd/sent.go:119]: Fixed in 1f7a2c2. The focused Markdown test now covers the compact recipient summary, app URL, and unavailable delivery time.

File: tests/smoke/sent_test.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	messages := dataAs[[]smokeSentMessage](t, resp)
+	for _, message := range messages {
+		if message.ID <= 0 {
+			t.Errorf("sent thread id = %d, want positive", message.ID)
+		}
+		if strings.TrimSpace(message.Subject) == "" {
+			t.Errorf("sent thread %d has no subject", message.ID)
+		}
+		if message.SentAt.IsZero() {
+			t.Errorf("sent thread %d has no sent_at", message.ID)
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply tests/smoke/sent_test.go]: Fixed in 1f7a2c2. The smoke shape now uses a nullable timestamp and rejects only a present zero value.

**Comment by robzolkos** [comment]
Fixed the latest review concerns in bc05bbf. Empty summaries now remain in JSON and Markdown, and a trimmed page with a continuation cursor no longer reports the fetched partial page as the total result count.

File: internal/cmd/sent.go [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+		}
+		if nextPage != "" {
+			opts = append(opts, output.WithMeta("next_page", nextPage))
+		}
+		return writeOK(messages, opts...)
+	}
+}
+
+func readSentPage(ctx context.Context, cursor string) (pageResult[generated.Topic], error) {
+	page, err := sdk.Topics().GetSentPage(ctx, cursor)
```
[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/cmd/sent.go:140]: Fixed in $(git rev-parse --short HEAD). API-COVERAGE.md now maps the Sent endpoint to Topics().GetSentPage and hey sent.

[the remark this answers is not in the document]
Comment (robzolkos) [reply internal/cmd/sent.go:140]: The fix is commit 75fe8f3.

**Comment by robzolkos** [comment]
Fixed - --all now reports the cursor after an empty continuation page, the styled timestamp test follows the local timezone, and sent_at falls back to created_at when active_at is unavailable.
